tests/introspection/tests.py

```diff
@@ -189,6 +184,15 @@ def test_get_indexes_multicol(self):
         self.assertNotIn('first_name', indexes)
         self.assertIn('id', indexes)
 
+    @skipUnlessDBFeature('can_introspect_index_type')
+    def test_get_constraints_indexes_types(self):
+        with connection.cursor() as cursor:
+            constraints = connection.introspection.get_constraints(cursor, Article._meta.db_table)
+        for key, val in constraints.items():
+            if val['index'] and not (val['primary_key'] or val['unique']):
```

Comment:
The `index_type = indexes['reporter_id'].pop('type', None)` pattern from the old test should be used rather than a loop -- otherwise, if there' some mistake in the if-condition or if the loop is empty, it's not certain that this assertion will ever run.